### PR TITLE
adds hapi-tracing

### PR DESCRIPTION
**Pull request by christopherjturner**

### CODE REVIEW

File: src/server/about/controller.js [hunk 1]
```diff
@@ -5,6 +5,7 @@
  */
 export const aboutController = {
   handler(_request, h) {
+    _request.logger.info('test')
```
Comment (feedmypixel) [inline src/server/about/controller.js:8]: so the underscore can be removed here. The underscore previously denoted the arg not being used. Now it is being used we can remove the underscore

File: src/server/common/helpers/logging/trace-id-mixin.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { getTraceId } from '@defra/hapi-tracing'
+
+export function traceIdMixin() {
```
Comment (feedmypixel) [inline src/server/common/helpers/logging/trace-id-mixin.js:3]: Not convinced on this abstraction. What if you want to add extra things in to the mixin? Considering the mixin pattern is a pino pattern IMO feels like this should just sit where it was in the loggerOptions. So its simple to see and override

**Review by feedmypixel — COMMENTED** [review]
A few comments - is it worth turning on SonarCloud to run this through?

File: jest.config.js [hunk 3]
```diff
@@ -21,7 +21,8 @@ export default {
     '<rootDir>/src/index.js',
     'index.js'
   ],
-  coverageDirectory: '<rootDir>/coverage'
+  coverageDirectory: '<rootDir>/coverage',
+  transformIgnorePatterns: ['/node_modules(?!/(@defra/hapi-tracing))']
```
Comment (feedmypixel) [inline jest.config.js:25]: comment maybe around why?

`// ESM only module`

File: src/server/about/controller.js (hunk 1 above)
Comment (christopherjturner) [reply src/server/about/controller.js]: the "test" log line isn't suppose to be in, keeping underscores and removing log

**Review by feedmypixel — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**